### ISSUE TITLE
Assert failure in error()

### ISSUE THREAD

**Issue by Zeex** [Southclaws/pawn#230]
Sample script:

```pawn
native print(const s[]);

#define d1\
    print("ok")
#define d2 \
    print("ok")

#warning this is\
    warning 1
#warning this is\
warning 2
#warning this is \
warning 3
#warning this is \
    warning 4

// single-line comments can span \
multiple lines if you really want it

#pragma deprecated don't\
    use \
this \
    function\
please
f() {}

main() {
    d1;
    d2;
    f();

    new s[] = "only one \
space here";
}
```

```
assert(errstart<=fline)
```

https://github.com/Southclaws/pawn/blob/79f38920d9c1ffecb4a86fb0aa9d2997dd3e6b28/source/compiler/sc5.c#L264

SA-MP compiler has this bug too (notice how error's starting line is greater than its ending line):

```
issue-217.pwn(35) : error 031: unknown directive
issue-217.pwn(35 -- 20) : error 049: invalid line continuation
issue-217.pwn(35 -- 27) : warning 234: function is deprecated (symbol "f") don'tuse this functionplease

issue-217.pwn(32) : warning 234: function is deprecated (symbol "f") don'tuse this functionplease

issue-217.pwn(35) : warning 204: symbol is assigned a value that is never used: "s"
Pawn compiler 3.2.3664	 	 	Copyright (c) 1997-2006, ITB CompuPhase


2 Errors.
```

It just has asserts disabled so no error popup there:

**Comment by YashasSamaga** [comment]
```
#pragma deprecated don't use please
f() {

}

main() {
  
}
```

no deprecated function warning which is right

```
#pragma deprecated don't use please
f() {

}

main() {
  f();
}
```

two warnings are thrown when just one should've been

**Comment by Zeex** [comment]
I agree, it should be reported only once at hat the point of use. But I think we need to create a separate issue for it.
